packages/notification-user-lifecycle-consumer/src/services/messageProcessor.ts

```diff
@@ -0,0 +1,182 @@
+import { notificationConfigApi } from "pagopa-interop-api-clients";
+import { UsersEventPayload } from "../model/UsersEventPayload.js";
+import { config } from "../config/config.js";
+import { ReadModelServiceSQL } from "./readModelServiceSQL.js";
+import { UserServiceSQL } from "./userServiceSQL.js";
+
+// eslint-disable-next-line max-params
+export async function processUserEvent(
+  payload: UsersEventPayload,
+  readModelServiceSQL: ReadModelServiceSQL,
+  userServiceSQL: UserServiceSQL,
+  notificationConfigProcessClient: ReturnType<
+    typeof notificationConfigApi.createProcessApiClient
+  >,
+  interopTokenGenerator: InteropTokenGenerator,
+  loggerInstance: ReturnType<typeof logger>,
+  correlationId: CorrelationId
+): Promise<void> {
+  const userId = unsafeBrandId<UserId>(payload.user.userId);
+  const institutionId = unsafeBrandId<SelfcareId>(payload.institutionId);
+  const action = payload.eventType;
+  const tenantId = await readModelServiceSQL.getTenantIdBySelfcareId(
+    institutionId
+  );
+
+  if (!tenantId) {
+    throw genericInternalError(
+      `Tenant not found for selfcareId: ${institutionId}`
+    );
+  }
+
+  const userData = {
+    userId,
+    tenantId,
+    institutionId,
+    name: payload.user.name,
+    familyName: payload.user.familyName,
+    email: payload.user.email,
+    productRole: payload.user.productRole,
+  };
+
+  // TODO try catch process call error
+  await match(action)
+    .with("add", async () => {
+      loggerInstance.info(`Add user id ${userId} from tenant ${tenantId}`);
+
+      try {
+        const { serialized } =
+          await interopTokenGenerator.generateInternalToken();
+        await notificationConfigProcessClient.createUserDefaultNotificationConfig(
+          {
+            userId,
+            tenantId,
+          },
+          {
+            headers: {
+              "X-Correlation-Id": correlationId,
+              Authorization: `Bearer ${serialized}`,
+            },
+          }
+        );
+      } catch (err) {
+        loggerInstance.error(
+          `Error creating user default notification config for user ${userId} from tenant ${tenantId}. Reason: ${err}`
+        );
+      }
+      return userServiceSQL.insertUser(userData);
+    })
+    .with("update", async () => {
+      loggerInstance.info(`Update user id ${userId} from tenant ${tenantId}`);
+      return userServiceSQL.updateUser(userData);
+    })
+    .with("delete", async () => {
+      loggerInstance.info(`Removing user ${userId} from tenant ${tenantId}`);
+      try {
+        const { serialized } =
+          await interopTokenGenerator.generateInternalToken();
+        await notificationConfigProcessClient.deleteUserNotificationConfig(
+          undefined,
+          {
+            params: {
+              userId,
+              tenantId,
+            },
+            headers: {
+              "X-Correlation-Id": correlationId,
+              Authorization: `Bearer ${serialized}`,
+            },
```

Comment:
in #2179 I used slightly different functions (a `RefreshableInteropToken` object and `getInteropHeaders`) because I took another consumer as a model... but it looks like consumers don't all follow the same pattern: anyway it looks mostly equivalent, but maybe judging from the comments in the code `RefreshableInteropToken` could be good to use to cache the token while it's valid